src/Runner.Worker/ExecutionContext.cs

```diff
@@ -254,7 +258,7 @@ public void RegisterPostJobStep(IStep step)
             DictionaryContextData inputsData,
             Dictionary<string, string> envData)
         {
-            step.ExecutionContext = Root.CreateChild(_record.Id, step.DisplayName, _record.Id.ToString("N"), scopeName, step.Action.ContextName, logger: _logger);
+            step.ExecutionContext = Root.CreateChild(_record.Id, step.DisplayName, _record.Id.ToString("N"), scopeName, step.Action.ContextName, logger: _logger, cancellationTokenSource: _cancellationTokenSource);
```

Comment:
Would it make more sense for the interface to pass `CancellationToken.CreateLinkedTokenSource(_cancellationTokenSource)`